.claude/hooks/smart_fake_code_detection.sh

```
@@ -0,0 +1,153 @@
+    else
+        RELATIVE_FILES+=("$path")
+    fi
+done
+
+# Sanitize file paths to prevent prompt injection via special characters
+sanitize_for_prompt() {
+    local path="$1"
+    path=${path//$'\n'/ }
+    path=${path//$'\r'/ }
+    path=${path//\\/\\\\}
+    path=${path//\`/\\`}
```

Shellcheck (0.11.0)

[error] Couldn't parse this backtick expansion. Fix to allow more checks.

(SC1073)